modules/queue/config.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package queue
+
+import (
+	"path/filepath"
+	"unicode"
+
+	"code.gitea.io/gitea/modules/setting"
+)
+
+type BaseConfig struct {
+	ManagedName string
+
+	DataFullDir string
+	ConnStr     string
+	Length      int
+
+	QueueFullName, SetFullName string
+}
+
+type IniConfig struct {
```

Comment:
I think we could have a better name. I don't think we should mention `ini` here. This package should not care whether is it from an `ini` format or file.

Reply:
It is. It comes from INI directly. See the `MapTo`.

We need to convert the INI config to our config.

Reply:
After reading from ini content or file, we should not mention `ini` in struct names. Otherwise, all structs in `setting` packages should be added a `ini` prefix?

Reply:
This package is not `setting` package.

Or could you suggest a new prefix you like?

Reply:
Is `BaseConfig` for a special queue and `IniConfig` for the global configuration of queue package?

Reply:
Yes

`IniConfig` also provides some options for the WorkerPoolQueue.

Reply:
> 

Why we don't use `setting.QueueSettings` directly?

Reply:
The new code is covered by tests.

`QueueSettings` was never tested.

Reply:
If you have strong preference to move the `IniConfig` to `setting` package, I can do that.

Reply:
Done in c99ee5d05464272880a9cf94a9bd83a19562b9ea  please help to review (mark this as resolved)